specs/language/conversions.tex

```diff
@@ -0,0 +1,135 @@
+\texttt{false} are converted to one and zero respectively.
+
+\Sec{Floating point conversion}{Conv.fconv}
+
+\p A glvalue of a floating point type can be converted to a cxvalue of any other
+floating point type. A prvalue of a floating point type can be converted to a
+prvalue of any other floating point type.
+
+\p If the source value can be exactly represented in the destination type, the
+conversion produces the exact representation of the source value. If the source
+value cannot be exactly represented, the conversion to a best-approximation of
+the source value is implementation defined.
+
+\Sec{Floating point-integral conversion}{Conv.fpint}
+
+\p A glvalue of floating point type can be converted to a cxvalue of integer
+type. A prvalue of floating point type can be converted to a prvalue of
+integer type. Conversion of floating point values to integer values truncates by
+discarding the fractional value. The behavior is undefined if the truncated
+value cannot be represented in the destination type.
+
+\p A glvalue of integer type can be converted to a cxvalue of floating point
+type. A prvalue of integer type can be converted to a prvalue of floating
+point type. If the destination type can exactly represent the source value, the
+result is the exact value. If the destination type cannot exactly represent the
+source value, the conversion to a best-approximation of the source value is
+implementation defined.
+
+\Sec{Boolean conversion}{Conv.bool}
+
+\p A glvalue of arithmetic type can be converted to a cxvalue of boolean type. A
+prvalue of arithmetic or unscoped enumeration type can be converted to a prvalue
+of boolean type. A zero value is converted to \texttt{false}; all other values
+are converted to \texttt{true}.
+
+\Sec{Vector splat conversion}{Conv.vsplat}
+
+\p A glvalue of type \texttt{T} can be converted to a cxvalue of type
+\texttt{vector<T,x>} or a prvalue of type \texttt{T} can be converted to a
+prvalue of type \texttt{vector<T,x>}. The destination value is the source value
+replicated into each element of the destination.
+
+\p A glvalue of type \texttt{T} can be converted to a cxvalue of type
+\texttt{matrix<T,x,y>} or a prvalue of type \texttt{T} can be converted to a
+prvalue of type \texttt{matrix<T,x,y>}. The destination value is the source
+value replicated into each element of the destination.
+
+\Sec{Vector and matrix truncation conversion}{Conv.vtrunc}
+
+\p A glvalue of type \texttt{vector<T,x>} can be converted to a cxvalue of type
+\texttt{vector<T,y>}, or a prvalue of type \texttt{vector<T,x>} can be converted
+to a prvalue of type \texttt{vector<T,y>} only if \texttt{x} is less than
+\texttt{y}.
+
+\p A glvalue of type \texttt{matrix<T,x,y>} can be converted to a cxvalue of type
+\texttt{matrix<T,z,w>}, or a prvalue of type \texttt{matrix<T,x,y>} can be
+converted to a prvalue of type \texttt{matrix<T,z,w>} only if \( x \leq z \)
+and \(y \leq w \)
```

Comment:
It might be helpful to have a brief code listing with examples for each section of conversions. Is that something that would be practical without it taking up too much space?